xrpl/models/requests/request.py

```diff
@@ -78,6 +84,66 @@ class Request(BaseModel):
     method: RequestMethod = REQUIRED  # type: ignore
     id: Optional[Union[str, int]] = None
 
+    @classmethod
+    def from_dict(cls: Type[Request], value: Dict[str, Any]) -> Request:
+        """
+        Construct a new Request from a dictionary of parameters.
+
+        Args:
+            value: The value to construct the Request from.
+
+        Returns:
+            A new Request object, constructed using the given parameters.
+
+        Raises:
+            XRPLModelException: If the dictionary provided is invalid.
+        """
+        if cls.__name__ == "Request":
+            if "method" not in value:
+                raise XRPLModelException("Request does not include method.")
+            correct_type = cls.get_method(value["method"])
+            return correct_type.from_dict(value)
+        else:
+            if "method" in value:
+                method = value["method"]
+                if _method_to_class_name(method) != cls.__name__ and not (
+                    method == "submit"
+                    and cls.__name__ in ("SignAndSubmit", "SubmitOnly")
+                ):
+                    raise XRPLModelException(
+                        f"Using wrong constructor: using {cls.__name__} constructor "
+                        f"with Request method {method}."
+                    )
+                value = {**value}
+                del value["method"]
+            return cast(Request, super(Request, cls).from_dict(value))
```

Comment:
nit - you can remove an indentation level from this already-far-indented `else` block by removing the `else`, since you `return` in the `if`